vms/platformvm/state/subnet_only_validator.go

```diff
@@ -22,9 +26,51 @@ var (
 	_ btree.LessFunc[SubnetOnlyValidator] = SubnetOnlyValidator.Less
 	_ utils.Sortable[SubnetOnlyValidator] = SubnetOnlyValidator{}
 
-	ErrMutatedSubnetOnlyValidator = errors.New("subnet only validator contains mutated constant fields")
+	ErrMutatedSubnetOnlyValidator     = errors.New("subnet-only validator contains mutated constant fields")
+	ErrConflictingSubnetOnlyValidator = errors.New("subnet-only validator contains conflicting subnetID + nodeID pair")
+	ErrDuplicateSubnetOnlyValidator   = errors.New("subnet-only validator contains duplicate subnetID + nodeID pair")
 )
 
+type SubnetOnlyValidators interface {
+	// GetActiveSubnetOnlyValidatorsIterator returns an iterator of all the
+	// active subnet-only validators in increasing order of EndAccumulatedFee.
+	GetActiveSubnetOnlyValidatorsIterator() (iterator.Iterator[SubnetOnlyValidator], error)
+
+	// NumActiveSubnetOnlyValidators returns the number of currently active
+	// subnet-only validators.
+	NumActiveSubnetOnlyValidators() int
+
+	// WeightOfSubnetOnlyValidators returns the total active and inactive weight
+	// of subnet-only validators on [subnetID].
+	WeightOfSubnetOnlyValidators(subnetID ids.ID) (uint64, error)
+
+	// GetSubnetOnlyValidator returns the validator with [validationID] if it
+	// exists. If the validator does not exist, [err] will equal
+	// [database.ErrNotFound].
```

Comment:
It's a bit odd to call out the error type here when that's not enforceable by the interface

Reply:
I don't think this is particularly unusual... This is done in the standard lib as well. For an example, see `SetDeadline` here: https://pkg.go.dev/net#Conn